atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/watch/ClientLogEvents.java

```diff
@@ -96,6 +97,33 @@ default CommitUpdate toCommitUpdate(LockWatchVersion startVersion, CommitInfo co
         return CommitUpdate.invalidateSome(locksTakenOut);
     }
 
+    default SpanningCommitUpdate toSpanningCommitUpdate(
+            LockWatchVersion startVersion, CommitInfo commitInfo, LockWatchVersion endVersion) {
+        if (clearCache()) {
+            return SpanningCommitUpdate.invalidateAll();
+        }
+
+        // We want to ensure that we do not miss any versions, but we do not care about the event with the same version
+        // as the start version.
+        verifyReturnedEventsEnclosesTransactionVersions(startVersion.version() + 1, endVersion.version());
+
+        LockEventVisitor eventVisitor = new LockEventVisitor(commitInfo.commitLockToken());
+        Set<LockDescriptor> transactionLocks = new HashSet<>();
```

Comment:
We can have a common method that returns a `Set` of something like `MaybeSpanningLockDescriptor`, from which you can easily construct one or the other so both of the methods can just call that. Takes an additional traversal for spanning but that's prob fine

Reply:
Alternatively, you can just have toCommitUpdate call toSpanning and take the spanning update (or the other one, really)